applications/print_array_elements.py

```diff
@@ -120,18 +120,18 @@ def _parse(label=None, description=None):
         default=False,
         action="store_true",
     )
-    return config.initialize()
+    return config.initialize(db_config=True)
 
 
 def main():
 
     label = Path(__file__).stem
-    args_dict, _ = _parse(label, description=("Print a list of array element positions"))
+    args_dict, db_config = _parse(label, description=("Print a list of array element positions"))
 
     _logger = logging.getLogger()
     _logger.setLevel(gen.get_log_level_from_user(args_dict["log_level"]))
 
-    layout = layout_array.LayoutArray()
+    layout = layout_array.LayoutArray(site="North", mongo_db_config=db_config)
```

Comment:
Good question. This is not the best solution. It would be better to get the site directly from the file (given by `args_dict["array_element_list"]`) that is anyways used in the line below this one. To do that we would have to read the file before initializing the class, therefore, I changed the code now and divided the `read_telescope_list_file` into two files: `read_telescope_list_file` (now static method, i.e. one can use without initializing the class) and `initialize_layout_array_from_telescope_file` which does exactly what `read_telescope_list_file` was doing before. With this we are able to implement a function that reads the metadata from the telescope file and give us the site, then we initialize the class. Nevertheless, in discussions with Orel we realized that we want to drastically reduce the metadata in the telescope file and instead of passing all the coordinates info, we simply pass the site and then the rest of the information is read from the DB. Since we will probably implement that later, I suggest we keep this "North" here and open an issue so we dont forget it.

Reply:
Do I understand it correctly that `site` is needed because now also `mongo_db_config` is added to the constructed, although it is only needed in `from_layout_array_name`? My suggestion would be to remove site and mongo_db_config from the `_init_` function. This means you wouldn't need the db config in print_array_elements.

Reply:
`mongo_db_config` will anyway be needed here cause we will always want to read the site info from the DB then from the file. So at least that we should keep.

Reply:
Maybe I misunderstand the changes - but in the current implementation `applications/print_array_elements.py` does not need the database, `LayoutArray:_init()_` does not need the DB - is only needed when calling `LayoutArray::from_layout_array_name`. So why adding a dependency, in the init function?

I think it is important to keep the code always as simple as possible - this is a tool used a lot when discussing telescope positions. Keeping the option of having all information in the metadata header of the ecsv files about sites is an important feature and should be kept.

Reply:
it seems to me we need to decide first if we want to get the site parameters from the DB or always from the header in the telescope files. Please see my question here:

#https://github.com/gammasim/gammasim-tools/issues/435#issuecomment-1354725181

Reply:
I was trying to remove the dependency on the site parameter, I found only one place where it is important now:
https://github.com/gammasim/gammasim-tools/blob/2e118f51f26bd53830035fcd46013cefbe6f94e0/simtools/layout/layout_array.py#L142

This function is now being used in the initialization of the class and it is important to collect the corsika parameters to build the array. especially the observational level is of importance here and it depends on which site we are talking about. To summarize, we need the site as a parameter to load the corsika parameters into the class instance.

I cant find a good solution now to get rid of the site parameter and at the same time keep the way we initialize the class here. Does any of you have an idea?